ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -134,20 +134,21 @@ let send_diagnostics ?diagnostics rpc doc =
               Query_protocol.Errors
                 { lexing = true; parsing = true; typing = true }
             in
-            let+ errors = Document.dispatch_exn doc command in
-            List.map errors ~f:(fun (error : Loc.error) ->
-                let loc = Loc.loc_of_report error in
-                let range = Range.of_loc loc in
-                let severity =
-                  match error.source with
-                  | Warning -> DiagnosticSeverity.Warning
-                  | _ -> DiagnosticSeverity.Error
-                in
-                let message =
-                  Loc.print_main Format.str_formatter error;
-                  String.trim (Format.flush_str_formatter ())
-                in
-                create_diagnostic range message ~severity)
+            Document.with_pipeline_exn doc (fun pipeline ->
+                let errors = Query_commands.dispatch pipeline command in
+                List.map errors ~f:(fun (error : Loc.error) ->
+                    let loc = Loc.loc_of_report error in
+                    let range = Range.of_loc loc in
+                    let severity =
+                      match error.source with
+                      | Warning -> DiagnosticSeverity.Warning
+                      | _ -> DiagnosticSeverity.Error
+                    in
+                    let message =
+                      Loc.print_main Format.str_formatter error;
+                      String.trim (Format.flush_str_formatter ())
```

Comment:
You can also try replacnig Format.str_formatter with Format.asprintf. I don't quite remember what the exact problem was.

Reply:
I just reverted changes to send_diagnostics because it's not really the goal of this PR. I don't quite understand what we want from experimenting with Format fns.